libsoftwarecontainer/src/gateway/dbus/dbusgateway.cpp

```diff
@@ -76,29 +87,26 @@ bool DBusGateway::activateGateway()
     bool hasEnvVar = false;
     std::string envValue = Glib::getenv(variable, hasEnvVar);
 
-    if (hasEnvVar) {
-        envVec.push_back(variable + "=" + envValue);
-    } else if (m_type == SessionProxy) {
-        log_error() << "Using DBus gateway in session mode"
-                    << " and no " + variable + " set in host environment, dbus-proxy won't work";
-        return false;
+    if (!hasEnvVar) {
+        if (m_type == SessionProxy) {
+            log_error() << "Using DBus gateway in session mode"
+                        << " and no " + variable + " set in host environment, dbus-proxy won't work";
+            return false;
+        } else {
+            log_warn() << "Using DBus gateway in system mode"
+                       << " and no " + variable + " set in host environment, this could be a problem";
+        }
     } else {
-        log_warn() << "Using DBus gateway in system mode"
-                   << " and no " + variable + " set in host environment, this could be a problem";
+        envVec.push_back(variable + "=" + envValue);
     }
 
     if (!startDBusProxy(commandVec, envVec)) {
         return false;
     }
 
-    // Write configuration
-    json_t *jsonConfig = json_object();
-    json_object_set(jsonConfig, DBusGatewayParser::SESSION_CONFIG, m_sessionBusConfig);
-    json_object_set(jsonConfig, DBusGatewayParser::SYSTEM_CONFIG, m_systemBusConfig);
-
-    char *config_c = json_dumps(jsonConfig, JSON_COMPACT);
+    // Dump to string and clean up
+    char *config_c = json_dumps(m_totConfig, JSON_COMPACT);
```

Comment:
You can use implicit conversion to std::string by declaring `std::string config` here directly, then you won't need to `to_string` it.